modules/src/ics02_client/msgs.rs

```diff
@@ -126,56 +130,109 @@ impl From<MsgCreateAnyClient> for RawMsgCreateClient {
             client_id: ics_msg.client_id.to_string(),
             client_state: Some(ics_msg.client_state.into()),
             consensus_state: Some(ics_msg.consensus_state.into()),
-            signer: ics_msg.signer.to_string(),
+            signer: account_to_string(ics_msg.signer).unwrap(),
```

Comment:
Yeah, this caused us some grief for nothing. I propose we remove the typed `signer` from IBC domain type structures as it is never really used there.  We should either remove it altogether or make it a `String`. But this should happen in a different issue, maybe together with @andynog's work on the `keystore`.
The only time this is written is when the relayer sends a message. The signer should be build from the `keystore`. When the relayer builds the domain type message it doesn't need the `signer`. The only place it is required after, is in the raw message created from the domain type message. The relayer could just add it the raw message at that point.
The place where the signer is read on the handlers side is in some future module that will do all the account, auth, etc. verification. This will happen before the IBC handlers are invoked and they make no use for the signer.